### PR TITLE
✨Bump go version to v1.20.4

### PR DESCRIPTION
**Pull request by furkatgofurov7**
**What this PR does / why we need it**:
Few CI jobs are failing in main branch ([e2e-main](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/directory/pull-cluster-api-operator-e2e-main/1679273348166586368), [test-main](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-operator/186/pull-cluster-api-operator-test-main/1679273348103671808)) and PRs like #186  due to go version incompatibility with k8s:

```
Activated service account credentials for: [prow-build@k8s-infra-prow-build.iam.gserviceaccount.com]
+ WRAPPED_COMMAND_PID=[2](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-operator/186/pull-cluster-api-operator-test-main/1679273348103671808#1:build-log.txt%3A2)7
+ wait 27
+ ./scripts/ci-test.sh
Detected go version: go version go1.19.10 linux/amd6[4](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-operator/186/pull-cluster-api-operator-test-main/1679273348103671808#1:build-log.txt%3A4).
Kubernetes requires go1.20.0 or greater.
Please install go1.20.0 or later.
+ EXIT_VALUE=2
+ set +o xtrace
```

This PR bumps go version (1st commit), fixes lint issue (2nd commit) and also allows users to override the builder image (3rd commit)

### CODE REVIEW

**Comment by furkatgofurov7** [comment]
hmm #187 already does take care of the go bump, I will rebase this once that is merged

/hold

File: cmd/main.go
Comment (Fedosin) [inline cmd/main.go:117]: Can we just remove this line? I don't see where we use random numbers in the project.

Comment (furkatgofurov7) [reply cmd/main.go]: Removed it completely

**Review by alexander-demicev — APPROVED** [review]
thanks!
/approve

**Pull request merged**
